### ISSUE TITLE
errobar interpretation

### ISSUE THREAD

**Issue by saassli21** [cafferychen777/ggpicrust2#69]
Hello, I'm new to picrust2 output analysis.
I'm wondering how to analyze this error bar, especially the log2 fold change section and the associated p-value!. Could please give me an example of intepretation [image](https://github.com/cafferychen777/ggpicrust2/assets/131655435/e8e9b207-26ea-410c-aa2e-0bbf99e62aa8)

**Comment by cafferychen777** [comment]
Hello,

When analyzing PICRUSt2 output, the log2 fold change is a measure used to express the change in abundance of features (such as genes or metabolic pathways) between two conditions. It is calculated by taking the logarithm base 2 of the ratio of the abundances. Specifically, if you have two conditions, A and B, and you want to calculate the fold change of a feature from A to B, you would divide the abundance in condition B by the abundance in condition A, and then take the log base 2 of that ratio. This is what the log2 fold change represents:

\[ \text{log2 fold change} = \log_2 \left( \frac{\text{Abundance in condition B}}{\text{Abundance in condition A}} \right) \]

A positive log2 fold change indicates an increase in abundance in condition B compared to A, while a negative value indicates a decrease.

Regarding the p-values and adjusted p-values (p_adjust), these are statistical measures that help to determine the significance of the observed changes. The p-value represents the probability that the observed change (or a more extreme change) could have occurred by random chance, under the null hypothesis of no true difference between conditions. Adjusted p-values (p_adjust) are used to correct for multiple comparisons, reducing the likelihood of false positives.

In your analysis, if you have filtered the results to retain only those with significant p_adjust values (typically < 0.05), then you should not expect to see pathways with p_adjust values greater than 0.05 in your plots. If you do see such cases, it might indeed be due to a display issue with ggplot. One potential solution is to round the p_adjust values before plotting to ensure that the significance threshold is properly applied. This can be done in R using the `round()` function, like so:

```R
daa_annotated_sub_method_results_df1$p_adjust <- round(daa_annotated_sub_method_results_df1$p_adjust, digits = significant_digits)
```

Where `significant_digits` is the number of decimal places you want to round to, which could be 2 or 3 depending on the level of precision you require. After rounding, you can re-check your plots to ensure that only significant results are displayed.

I hope this clarifies your concerns. If you continue to encounter unexpected results, it may be worth revisiting your filtering criteria or the code used to generate the plots to ensure everything is functioning as intended.

**Comment by saassli21** [comment]
> Hello,
> 
> When analyzing PICRUSt2 output, the log2 fold change is a measure used to express the change in abundance of features (such as genes or metabolic pathways) between two conditions. It is calculated by taking the logarithm base 2 of the ratio of the abundances. Specifically, if you have two conditions, A and B, and you want to calculate the fold change of a feature from A to B, you would divide the abundance in condition B by the abundance in condition A, and then take the log base 2 of that ratio. This is what the log2 fold change represents:
> 
> [ \text{log2 fold change} = \log_2 \left( \frac{\text{Abundance in condition B}}{\text{Abundance in condition A}} \right) ]
> 
> A positive log2 fold change indicates an increase in abundance in condition B compared to A, while a negative value indicates a decrease.
> 
> Regarding the p-values and adjusted p-values (p_adjust), these are statistical measures that help to determine the significance of the observed changes. The p-value represents the probability that the observed change (or a more extreme change) could have occurred by random chance, under the null hypothesis of no true difference between conditions. Adjusted p-values (p_adjust) are used to correct for multiple comparisons, reducing the likelihood of false positives.
> 
> In your analysis, if you have filtered the results to retain only those with significant p_adjust values (typically < 0.05), then you should not expect to see pathways with p_adjust values greater than 0.05 in your plots. If you do see such cases, it might indeed be due to a display issue with ggplot. One potential solution is to round the p_adjust values before plotting to ensure that the significance threshold is properly applied. This can be done in R using the `round()` function, like so:
> 
> ```r
> daa_annotated_sub_method_results_df1$p_adjust <- round(daa_annotated_sub_method_results_df1$p_adjust, digits = significant_digits)
> ```
> 
> Where `significant_digits` is the number of decimal places you want to round to, which could be 2 or 3 depending on the level of precision you require. After rounding, you can re-check your plots to ensure that only significant results are displayed.
> 
> I hope this clarifies your concerns. If you continue to encounter unexpected results, it may be worth revisiting your filtering criteria or the code used to generate the plots to ensure everything is functioning as intended.

thank you very much for these clarification. So if we take the example of #65. the user published this figure 
![image](https://github.com/cafferychen777/ggpicrust2/assets/131655435/90856d67-7210-4851-b926-99b9e29b27ed)
 based on the script that you suggested but by applying limma test , he generates p-adjust value >0.05 on the plot although he filtered only those with p-adjust < 0.05 previously. in this context he must apply round() function ?
- otherwise can we apply round() function by default not necessarily only when we observe p-adjust > 0.05

**Comment by cafferychen777** [comment]
Hello,

Thank you for following up with additional details and the specific example.

Yes, applying the `round()` function in the context you've described can indeed help address the issue where the p-adjust values greater than 0.05 are appearing in plots despite previous filtering for values less than 0.05. This can happen due to the way ggplot handles numeric precision during plotting.

However, regarding your question about applying the `round()` function by default: while this can be a useful approach, it's important to consider that rounding by default may not always be appropriate. For instance, in cases where the p-adjust values already have a low number of decimal places, default rounding could inadvertently alter the data's precision and potentially impact the accuracy of the visual representation. 

I had tested this approach previously and found that it could indeed influence the visualization results in certain scenarios. Therefore, I decided not to set default rounding in the package to avoid unintended alterations to the data.

In summary, applying the `round()` function can be a valuable step when encountering issues with p-adjust values in plots. However, whether to apply it by default should be carefully considered based on the specific context and nature of the data being visualized.

I hope this helps clarify the reasoning behind the package's design choices. If you have any more questions or need further assistance, feel free to reach out.

Best regards,
Chen YANG

**Comment by cafferychen777** [comment]
Hello,

Thank you for following up with additional details and the specific example.

Yes, applying the `round()` function in the context you've described can indeed help address the issue where the p-adjust values greater than 0.05 are appearing in plots despite previous filtering for values less than 0.05. This can happen due to the way ggplot handles numeric precision during plotting.

However, regarding your question about applying the `round()` function by default: while this can be a useful approach, it's important to consider that rounding by default may not always be appropriate. For instance, in cases where the p-adjust values already have a low number of decimal places, default rounding could inadvertently alter the data's precision and potentially impact the accuracy of the visual representation. 

I had tested this approach previously and found that it could indeed influence the visualization results in certain scenarios. Therefore, I decided not to set default rounding in the package to avoid unintended alterations to the data.

In summary, applying the `round()` function can be a valuable step when encountering issues with p-adjust values in plots. However, whether to apply it by default should be carefully considered based on the specific context and nature of the data being visualized.

I hope this helps clarify the reasoning behind the package's design choices. If you have any more questions or need further assistance, feel free to reach out.

Best regards,
Chen YANG

**Comment by saassli21** [comment]
> Hello,
> 
> Thank you for following up with additional details and the specific example.
> 
> Yes, applying the `round()` function in the context you've described can indeed help address the issue where the p-adjust values greater than 0.05 are appearing in plots despite previous filtering for values less than 0.05. This can happen due to the way ggplot handles numeric precision during plotting.
> 
> However, regarding your question about applying the `round()` function by default: while this can be a useful approach, it's important to consider that rounding by default may not always be appropriate. For instance, in cases where the p-adjust values already have a low number of decimal places, default rounding could inadvertently alter the data's precision and potentially impact the accuracy of the visual representation.
> 
> I had tested this approach previously and found that it could indeed influence the visualization results in certain scenarios. Therefore, I decided not to set default rounding in the package to avoid unintended alterations to the data.
> 
> In summary, applying the `round()` function can be a valuable step when encountering issues with p-adjust values in plots. However, whether to apply it by default should be carefully considered based on the specific context and nature of the data being visualized.
> 
> I hope this helps clarify the reasoning behind the package's design choices. If you have any more questions or need further assistance, feel free to reach out.
> 
> Best regards, Chen YANG

Thank you very much for this detailed explanation! I appreciate it.
